src/ImageSharp.Drawing/Processing/Processors/Drawing/DrawImageProcessor.cs

```diff
@@ -81,6 +81,12 @@ protected override void OnFrameApply(ImageFrame<TPixelDst> source, Rectangle sou
 
             var workingRect = Rectangle.FromLTRB(minX, minY, maxX, maxY);
 
+            // not a valid operation because rectangle does not overlap with this image.
+            Guard.IsFalse(

```

Comment:
In my opinion, we only use `Guard` for trivial argument validation at the beginning of the methods. 

I think this is what we should do:
- Throw an `ImageProcessingException` if the condition is not matched
- In `ImageProcessor<T>.Apply(...)` we should detect (catch and rethrow) `ImageProcessingException`

@dlemstra @JimBobSquarePants thoughts?

Reply:
I think this check is superfluous and the errors in `ParallelHelper.ValidateRectangle` should be `ArgumentOutOfRangeException` via `Guard.MustBeGreaterThan`.

We already capture the inner exception and throw an `ImageProcessingException` in release mode. 

@atruskie sorry to be messing you around here.

Reply:
My comment was actually addressing a concern from @atruskie:
>I 'm pretty unhappy with the gymnastics required to expect an exception from an ImageProcessor. Any suggestions?

However, the more I think about fixing this, the further it leads, so let's not waste time on it in this PR.

Reply:
@antonfirsov I've followed your advice here already, and it leads to a nicer test.

@JimBobSquarePants I'd like to leave both errors in because they represent different types of error. One is descriptive for a user mistake. The other, in IterRows, really is exceptional and represents that that code should never have been called with the given argument.

Reply:
I’m trying to avoid as much as possible, per-processor error handling. A negative rectangle would cause an error in many places and multiple handlers would have to be added. 

I do wonder whether we should be using Recangle.Intersect to avoid the possibility as we do in several places?

Reply:
Sorry, I didn't see this last message before I pushed.

I feel like I'm not doing a good job at understanding what you'd like. The options:

1. Throw in DrawImage if there is not an overlap with a descriptive error message (my preference)
2. Have no-effect in DrawImage if there is not an overlap (my original version of this fix, possibly what you mean by doing Rectangle.Intersect checks)?
3. Rely on the exception in IterRows when a non-overlap occurs (I think this is only marginally better than the original confusing error message this PR is seeking to address)
4. Some other option?

I'm happy to keep making changes until I get this right :-) , it just feels a little like walking in circles.

Reply:
In retrospect I think you're doing the correct thing for now with the API we have. I'm going to update and merge this. Thanks for your help! 😄